packages/lib/src/helpers/encodeCall.ts

```diff
@@ -1,15 +1,35 @@
 import abi from 'ethereumjs-abi';
 import BN from 'bignumber.js';
 
-function formatValue(value: any): string {
-  if (typeof(value) === 'number' || BN.isBigNumber(value)) return value.toString();
-  else if (typeof(value) === 'string' && value.match(/\d+(\.\d+)?e(\+)?\d+/)) return (new BN(value)).toString(10);
-  else return value;
+export function formatValue(value: any): string | void {
+
+  // Numbers.
+  if(typeof(value) === 'number') {
+    if(value % 1 === 0) return value.toString(10);
+    else throw new Error('Floating point numbers are not supported on parameter encoding.');
+  }
+
+  // Big numbers.
+  if(BN.isBigNumber(value)) return value.toString();
+
+  // Strings.
+  if(typeof(value) === 'string') {
+    const hasHexRadix = value.indexOf('0x') !== -1 || value.indexOf('0X') !== -1;
```

Comment:
Shouldn't we check that the 0x radix is at the beginning of the string (maybe ignoring whitespace)? Otherwise "hell0x" whould test positive for having a 0x radix.

Reply:
Can we add a test for this, to make sure that "hell0xa" is handled properly as a string?